File: app/src/main/java/com/mdmobile/pocketconsole/utils/BaseActivity.java
package com.mdmobile.pocketconsole.utils;

import android.content.Intent;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.mdmobile.pocketconsole.ui.logIn.LoginActivity;

/**
 * Every Activity should extend this base Activity
 * It handles:
 * 1. checks for account validity
 */

public class BaseActivity extends AppCompatActivity {


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState, @Nullable PersistableBundle persistentState) {
        super.onCreate(savedInstanceState, persistentState);
        if (!checkActiveUser()) {
            redirectToLoginPage();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (!checkActiveUser()) {
            redirectToLoginPage();
        }
    }

    private boolean checkActiveUser() {
        return UserUtility.checkAnyUserLoggedIn();
    }

    private void redirectToLoginPage(){
        Intent intent = new Intent(this, LoginActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/mdmobile/pocketconsole/ui/BaseActivity.kt
package com.mdmobile.pocketconsole.ui

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import android.support.annotation.IdRes
import android.support.v7.app.AppCompatActivity
import android.view.View

import com.mdmobile.pocketconsole.ui.logIn.LoginActivity
import com.mdmobile.pocketconsole.utils.UserUtility

/**
 * Every Activity should extend this base Activity
 * It handles:
 * 1. checks for account validity
 */

open class BaseActivity : AppCompatActivity() {

    //Extension function to add find view by id in Activity
    fun <T : View> Activity.bind(@IdRes id: Int): Lazy<T> {
        return lazy(LazyThreadSafetyMode.NONE) { findViewById<T>(id) }
    }

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        if (!checkActiveUser()) {
            redirectToLoginPage()
        }
    }

    override fun onResume() {
        super.onResume()
        if (!checkActiveUser()) {
            redirectToLoginPage()
        }
    }

    private fun checkActiveUser(): Boolean {
        return UserUtility.checkAnyUserLoggedIn()
    }

    private fun redirectToLoginPage() {
        val intent = Intent(this, LoginActivity::class.java)
        startActivity(intent)
    }
}
